tests/check/check.go

```diff
@@ -196,3 +199,430 @@ func runTest(t *testing.T, test individualTest, params testParams, contextTupleT
+					Expectation: false,
+				},
+				{
+					Name:        "valid_cond_valid_user",
+					Tuple:       &openfgav1.TupleKey{Object: "directs-user:1", Relation: "direct_wild_cond", User: "user:valid"},
+					Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+					Expectation: true,
+				},
+				{
+					Name:        "invalid_cond_valid_user",
+					Tuple:       &openfgav1.TupleKey{Object: "directs-user:1", Relation: "direct_wild_cond", User: "user:valid"},
+					Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("2")}},
+					Expectation: false,
+				},
+				{
+					Name:        "valid_cond_invalid_user",
+					Tuple:       &openfgav1.TupleKey{Object: "directs-user:1", Relation: "direct_wild_cond", User: "user:invalid"},
+					Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+					Expectation: true,
+				},
+				{
+					Name:        "invalid_cond_invalid_user",
+					Tuple:       &openfgav1.TupleKey{Object: "directs-user:1", Relation: "direct_wild_cond", User: "user:invalid"},
+					Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("2")}},
+					Expectation: false,
+				},
+				{
+					Name:        "valid_cond_not_public_id",
+					Tuple:       &openfgav1.TupleKey{Object: "directs-user:not-public", Relation: "direct_wild_cond", User: "user:valid"},
+					Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+					Expectation: false,
+				},
+				{
+					Name:        "invalid_cond_not_public_id",
+					Tuple:       &openfgav1.TupleKey{Object: "directs-user:not-public", Relation: "direct_wild_cond", User: "user:valid"},
+					Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("2")}},
+					Expectation: false,
+				},
+			},
+		},
+	},
+}
+
+func runTestMatrix(t *testing.T, params testParams) {
+	schemaVersion := params.schemaVersion
+	client := params.client
+	name := matrix.Name
+
+	ctx := context.Background()
+	resp, err := client.CreateStore(ctx, &openfgav1.CreateStoreRequest{Name: name})
+	require.NoError(t, err)
+	storeID := resp.GetId()
+
+	var modelID string
+	t.Run(name, func(t *testing.T) {
+		for _, stage := range matrix.Stages {
+			t.Run(fmt.Sprintf("stage_%s", stage.Name), func(t *testing.T) {
+				if stage.Model != "" {
+					model := testutils.MustTransformDSLToProtoWithID(stage.Model)
+					writeModelResponse, err := client.WriteAuthorizationModel(ctx, &openfgav1.WriteAuthorizationModelRequest{
+						StoreId:         storeID,
+						SchemaVersion:   schemaVersion,
+						TypeDefinitions: model.GetTypeDefinitions(),
+						Conditions:      model.GetConditions(),
+					})
+					require.NoError(t, err)
+					modelID = writeModelResponse.GetAuthorizationModelId()
+				}
+
+				tuples := stage.Tuples
+				tuplesLength := len(tuples)
+				if tuplesLength > 0 {
```

Comment:
nit: I don't think the `if` condition here is necessary since the loop will never enter if `tuplesLength` is `< 1`.